torch/_functorch/_aot_autograd/traced_function_transforms.py

```diff
@@ -329,6 +329,20 @@ def traced_forward(*primals_fwd_seed_fwd_base_offset):
         return traced_forward, (*args, fwd_seed, fwd_base_offset)
 
 
+@contextmanager
+def set_subgraph(subgraph: str):
+    meta_key = "subgraph"
+    if fx_traceback.has_preserved_node_meta():
```

Comment:
This is saying that if `has_preserved_node_meta() == False`, we will just silently not add any metadata here (and fail to tell the partitioner that our mutation was a backward mutation, giving us wrong results).

Since we're relying on metadata for this correctness, we should probably just assert that this is true?